tests/unit/Core/Framework/DataAbstractionLayer/ExtensionRegistryTest.php

```diff
@@ -0,0 +1,85 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Tests\Unit\Core\Framework\DataAbstractionLayer;
+
+use PHPUnit\Framework\Attributes\CoversClass;
+use PHPUnit\Framework\TestCase;
+use Shopware\Core\Content\Category\CategoryDefinition;
+use Shopware\Core\Content\Product\ProductDefinition;
+use Shopware\Core\Framework\DataAbstractionLayer\BulkEntityExtension;
+use Shopware\Core\Framework\DataAbstractionLayer\Dbal\EntityWriteGateway;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityDefinition;
+use Shopware\Core\Framework\DataAbstractionLayer\ExtensionRegistry;
+use Shopware\Core\Framework\DataAbstractionLayer\Field\FkField;
+use Shopware\Core\Framework\DataAbstractionLayer\Field\ManyToOneAssociationField;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Test\Stub\DataAbstractionLayer\StaticDefinitionInstanceRegistry;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+
+/**
+ * @internal
+ */
+#[Package('core')]
+#[CoversClass(ExtensionRegistry::class)]
+class ExtensionRegistryTest extends TestCase
+{
+    public function testNoBulkRegistered(): void
+    {
+        $registry = new ExtensionRegistry([], []);
+
+        $bulks = $registry->buildBulkExtensions(
+            $this->registry([])
+        );
+
+        static::assertEmpty($bulks);
+    }
+
+    public function testUnkownDefinition(): void
+    {
+        $registry = new ExtensionRegistry([], [new MyBulkExtension()]);
+
+        $bulks = $registry->buildBulkExtensions(
+            $this->registry([ProductDefinition::class])
+        );
+
+        static::assertCount(1, $bulks);
+    }
+
+    public function testAllDefinitionsFound(): void
+    {
+        $registry = new ExtensionRegistry([], [new MyBulkExtension()]);
+
+        $bulks = $registry->buildBulkExtensions(
+            $this->registry([ProductDefinition::class, CategoryDefinition::class])
```

Comment:
the test with an attribute entity definition is still missing right?

The static registry should already support those definitions as well via
```php
['attribute_entity.definition' => new AttributeEntityDefinition(...)]
```

So it should be possible to cover that case as well in this test setup right?